### PR TITLE
python: add tests for railjson_generator

### LINKED ISSUE #6221: Check Python tests coverage
### Description and goal

Part of our current Python/CI effort

### Acceptance criteria

TBD

### PR DESCRIPTION
**Pull request by shenriotpro**
fixes https://github.com/osrd-project/osrd/issues/6221

### CODE REVIEW

File: python/railjson_generator/README.md [hunk 1]
```diff
@@ -82,3 +88,25 @@ Route can either be manually created, or generated using `generate_routes`, and
...
+$ poetry run pflake8 --config ./pyproject.toml
+$ poetry run pytype -j auto
+```
+
+Use black and isort to fix formatting.
+
+```sh
+$ poetry run black .
+$ poetry run isort .
+```
```
Comment (Khoyo) [inline python/railjson_generator/README.md]: Missing newline at EOF

File: python/railjson_generator/railjson_generator/external_generated_inputs.py [hunk 2]
```diff
@@ -28,14 +32,17 @@ class ExternalGeneratedInputs:
     electrical_profiles: List[ElectricalProfile] = field(default_factory=list)
 
     def add_electrical_profile(self, *args, **kwargs) -> ElectricalProfile:
+        """Builds an electrical profile, adds it to the inputs, and returns it."""
         self.electrical_profiles.append(ElectricalProfile(*args, **kwargs))
         return self.electrical_profiles[-1]
 
-    def save(self, path):
+    def save(self, path: PathLike):
+        """Writes to the path as a railjson."""
```
Comment (Khoyo) [inline python/railjson_generator/railjson_generator/external_generated_inputs.py]: Raijson is a file format, drop the `a`

**Review by Khoyo — APPROVED** [review]
Otherwise LGTM.

As an aside, we should probably change the docstrings to an imperative tense (https://peps.python.org/pep-0257/#one-line-docstrings)

**Comment by shenriotpro** [comment]
> Otherwise LGTM.
> 
> As an aside, we should probably change the docstrings to an imperative tense (https://peps.python.org/pep-0257/#one-line-docstrings)

Done, thanks.

**Review by Tguisnet — APPROVED** [review]
Thanks, good PR !
I have only one thing to say, the ascii diagrams giving details about the infra used for the tests are good.
However, for a better understanding, I think we could represent a switch with 'o' or something to have a better understanding of the infra.
Especially in the case for routes tests, to avoid thinking detectors are link between track sections.
And I have no particuliar suggestion, but the difference between the diagram crossing and the diagram double slip switch is not explicit too.

For the routes tests, maybe we should add a comment to know what kind of route we are trying to create, but it's not mandatory, so as you wish !

**Comment by shenriotpro** [comment]
> Thanks, good PR ! I have only one thing to say, the ascii diagrams giving details about the infra used for the tests are good. However, for a better understanding, I think we could represent a switch with 'o' or something to have a better understanding of the infra. Especially in the case for routes tests, to avoid thinking detectors are link between track sections. And I have no particuliar suggestion, but the difference between the diagram crossing and the diagram double slip switch is not explicit too.
> 
> For the routes tests, maybe we should add a comment to know what kind of route we are trying to create, but it's not mandatory, so as you wish !

Thanks, I made a few adjustments.

**Review by ghost — APPROVED** [review]
Reviewed the CI part which looks good to me

**Pull request merged**